aptos-move/block-executor/src/txn_last_input_output.rs

```diff
@@ -226,18 +226,30 @@ impl<K: ModulePath, T: TransactionOutput, E: Debug + Send + Clone> TxnLastInputO
     }
 
     /// Returns the total gas, execution gas, io gas and storage gas of the transaction.
-    pub fn fee_statement(&self, txn_idx: TxnIndex) -> Option<FeeStatement> {
+    pub(crate) fn fee_statement(&self, txn_idx: TxnIndex) -> Option<FeeStatement> {
         match &self.outputs[txn_idx as usize]
             .load_full()
             .expect("[BlockSTM]: Execution output must be recorded after execution")
             .output_status
         {
-            ExecutionStatus::Success(output) => Some(output.fee_statement()),
+            ExecutionStatus::Success(output) | ExecutionStatus::SkipRest(output) => {
```

Comment:
Why there was no `ExecutionStatus::SkipRest(output)` before?

Reply:
minibug, we weren't counting the fee from the last txn (because skip is after this one). It wouldn't affect what's included or not and thus the state, because the txn that gets above the gas limit also gets included, but it would potentially diverge on the metrics.